### DISCUSSION TITLE
antd 全局国际化不生效问题

### DISCUSSION THREAD

**Discussion by wangsanshao** [umijs/umi#12360] [Q&A] [answered]
antd 5 umi4。这样子配置antd全局国际化，不生效
```
// app.tsx
import { ConfigProvider } from 'antd'
import zh_CN from 'antd/locale/zh_CN'

export function rootContainer(container: any) {
  return <ConfigProvider locale={zh_CN}>
    {container}
 </ConfigProvider>
}
```

**Comment by fz6m** [comment, marked as answer]
这要看你用没用 umi 的 antd 插件，umi 的 antd 插件在某些配置下，会自己加 ConfigProvider ，在运行时配置可以改这个 ConfigProvider 的配置，详见 [antd 运行时配置](https://umijs.org/docs/max/antd#%E8%BF%90%E8%A1%8C%E6%97%B6%E9%85%8D%E7%BD%AE) 。

另外还要看你用没用 locale 国际化插件，因为国际化插件和 antd 一起用默认开了国际化。

据我所知这并不能保证所有组件都被国际化，有人反馈过某个组件还要单独传入 locale 配置（我忘了是哪个）。

因为干扰的因素很多，要不你就自己研究下，要不就给一个最小复现看看。
